### PR TITLE
feat: add state param to oauth calls

### PR DESCRIPTION
**Pull request by Thibault-Brocheton**
Revive PR https://github.com/tolgee/tolgee-platform/pull/1625

Does fix my issue https://github.com/tolgee/tolgee-platform/issues/1830

### CODE REVIEW

**Comment by JanCizmar** [comment]
Thanks! Looks fine 🚀 Let's wait for the tests. 👀

**Comment by Thibault-Brocheton** [comment]
I had to upgrade typescript due to the frontend static check failing (randomUUID is in typescript >= 4.6)

**Comment by Thibault-Brocheton** [comment]
That should works fine, tsc and e2e works locally

**Comment by Thibault-Brocheton** [comment]
* Fixed prettier / eslint errors
* Added babel dependencie to avoid warning message during build
* Upgraded eslint to avoid warning message when running eslint (complaining about the typescript upgrade)
* removed eslint annotations in tolgee.<env>.ts cause eslint complains about them (says they are useless)

**Comment by Thibault-Brocheton** [comment]
oupsy for eslint, didn't run the e2e one
fot the e2e fail I feel like they are flacky

**Comment by Thibault-Brocheton** [comment]
image
e2e ran find this time in local

**Comment by Thibault-Brocheton** [comment]
Hey @JanCizmar, do you have such flakyness in other PR, or is it linked to some package upgrade on my side ?
Are you able to restart the specific failing e2e ?

**Comment by JanCizmar** [comment]
I never seen this one failing randomly, but it looks like it's unrelated what you have done, so let's see what happens on second try.

File: webapp/src/service/AnonymousIdService.ts [hunk 1]
```diff
@@ -12,7 +12,6 @@ export const AnonymousIdService = {
   reset() {
     return localStorage.setItem(
       ANONYMOUS_ID_LOCAL_STORAGE_KEY,
-      // @ts-ignore
       crypto.randomUUID()
```
Comment (scndel) [inline webapp/src/service/AnonymousIdService.ts:15]: Logging has disappeared from #1625 tests but, in my memories, there was indeed an issue with randomUUID.

**Review by scndel — APPROVED** [review]

**Comment by JanCizmar** [comment]
Hey! According to the @cyyynthia [comment](
https://github.com/tolgee/tolgee-platform/issues/1830#issuecomment-1658305341), this won't work right? Do I understand it correctly? 

Sorry, I am currently very busy, so I cannot deep dive into it and I am not able to tell whether it's OK to merge it or not.

File: webapp/src/service/AnonymousIdService.ts (hunk 1 above)
Comment (Thibault-Brocheton) [reply webapp/src/service/AnonymousIdService.ts:15]: yeah that seems related to why I upgraded typescript, so I assumed I could remove this ts ignore

**Comment by Thibault-Brocheton** [comment]
i'm not sure I fully understand what @cyyynthia is trying to say about the "sub" parameter, but I can confirm this PR works fine, I tested it in local before submitting it with my Oauth provider (Jumpcloud)
I'm able to connect to tolgee with me account correctly created

**Comment by cyyynthia** [comment]
Then the payload sent from Jumpcloud does send the user with the format expected by the server. I'd still double check if the external user id is correctly detected (and that the Tolgee server correctly interprets the user data it fetched from Jumpcloud), otherwise then it seems acceptable to merge as-is. 👍

**Comment by Thibault-Brocheton** [comment]
From login page, I click on Oauth Login
image
On my provider page, I input my credentials
image
I'm redirected on project page. On my user profile, my fullname & email is correct.
Capture d’écran du 2023-08-02 16-31-04

**Comment by cyyynthia** [comment]
I meant more specifically the external user id that Tolgee stores in the database. It's important this data is understood correctly and isn't using irrelevant data which could lead to link conflicts and improper grant of authorization. If this data is correct, then I don't have any objections to merge.

**Comment by Thibault-Brocheton** [comment]
I'm trying to debug the application but i can't make my breakpoints work :( 
I checked conent of userResponse?.sub in :
https://github.com/tolgee/tolgee-platform/blob/869dfaf2ccbf43f66b5c82b03b37bde15078b02d/backend/api/src/main/kotlin/io/tolgee/security/third_party/OAuth2Delegate.kt#L81-L91

"sub" is the ID of my user in jumpcloud in their system (confirmed by looking in my user page url in their website)
"email" is my associated email

Then you have this line `newUserAccount.thirdPartyAuthId = userResponse.sub`


So it seems my Jumcloud ID is well registered in your database

**Comment by cyyynthia** [comment]
Perfect then. Green light from me 👍

**Comment by JanCizmar** [comment]
OK! I am hittin it!

**Review by JanCizmar — APPROVED** [review]

**Comment by scndel** [comment]
Thanks @Thibault-Brocheton  @JanCizmar @cyyynthia  ! Our _Tolgee_ now works fine with **_Okta_** 🎉

**Pull request merged**